src/gui/sdlmain.cpp

```diff
@@ -4639,6 +4651,75 @@ void GFX_GetSize(int &width, int &height, bool &fullscreen)
 	fullscreen = sdl.desktop.fullscreen;
 }
 
+static bool check_env_for_high_dpi_handling()
+{
+	constexpr auto disable_high_dpi_var = "SDL_HINT_VIDEO_HIGHDPI_DISABLED";
```

Comment:
Are you sure you're not doing what SDL already does? Hints are supposed to take precedence over flags, if I remember correctly.

Reply:
Also, there are `SDL_GetHint()` and `SDL_GetHintBoolean()` methods available.